src/loki/search.cc

```diff
@@ -57,6 +57,13 @@ bool side_filter(const PathLocation::PathEdge& edge, const Location& location, G
   auto* node = reader.GetEndNode(opp, tile);
   if (!node)
     return false;
+
+  // nothing to filter if it is a minor road
+  // since motorway = 0 and service = 7, higher number means smaller road class
+  uint32_t road_class = static_cast<uint32_t>(opp->classification());
+  if (road_class > location.street_side_cutoff_)
+    return false;
```

Comment:
tiny nit: could put that right after getting the `opp`, since we don't need the node for this